### PR TITLE
Remove obsolete platforms ifdefs from PAL

### PR DESCRIPTION
**Pull request by janvorli**
There were many ifdefs for obsolete platforms and processors like AIX, HPUX,
SPARC, ALPHA etc. This change is a cleanup that removes them.

### CODE REVIEW

**Comment by janvorli** [comment]
@jkotas can you take a look please?

File: src/pal/tests/palsuite/composite/synchronization/nativecs_interlocked/pal_composite_native_cs.cpp [hunk 1]
```diff
@@ -449,12 +449,8 @@ ULONGLONG GetTicks(void)
     asm volatile("rdtsc":"=a" (a), "=d" (d));
     return ((ULONGLONG)((unsigned int)(d)) << 32) | (unsigned int)(a);
 #else
-#if defined(__sparc__) || (defined (_HPUX_) && defined(__ia64__))
-    return (ULONGLONG)gethrtime();
-#else
     // #error Don''t know how to get ticks on this platform
     return (ULONGLONG)gethrtime();
-#endif // __sparc__
 #endif // _X86_
```
Comment (adityamandaleeka) [inline src/pal/tests/palsuite/composite/synchronization/nativecs_interlocked/pal_composite_native_cs.cpp]: Should be i386

File: src/pal/tests/palsuite/composite/synchronization/nativecriticalsection/pal_composite_native_cs.cpp [hunk 2]
```diff
@@ -445,12 +445,8 @@ ULONGLONG GetTicks(void)
     asm volatile("rdtsc":"=a" (a), "=d" (d));
     return ((ULONGLONG)((unsigned int)(d)) << 32) | (unsigned int)(a);
 #else
-#if defined(__sparc__) || (defined (_HPUX_) && defined(__ia64__))
-    return (ULONGLONG)gethrtime();
-#else
     // #error Don''t know how to get ticks on this platform
     return (ULONGLONG)gethrtime();
-#endif // __sparc__
 #endif // _X86_
```
Comment (adityamandaleeka) [inline src/pal/tests/palsuite/composite/synchronization/nativecriticalsection/pal_composite_native_cs.cpp]: i386?

**Comment by adityamandaleeka** [comment]
LGTM 🎉

File: src/pal/src/file/disk.cpp [hunk 3]
```diff
@@ -26,10 +26,8 @@ Revision History:
 #include "pal/stackstring.hpp"
 
 #include <sys/param.h>
-#if !defined(_AIX)
 // do we actually need this on other platforms. We don't seem to be using anything from there
```
Comment (jkotas) [inline src/pal/src/file/disk.cpp]: Nit delete this comment

File: src/pal/src/file/file.cpp [hunk 4]
```diff
@@ -3657,12 +3657,10 @@ DWORD FILEGetLastErrorFromErrno( void )
     case EEXIST:
         dwRet = ERROR_ALREADY_EXISTS; 
         break;
-#if !defined(_AIX)
     // ENOTEMPTY is the same as EEXIST on AIX. Meaningful when involving directory operations
```
Comment (jkotas) [inline src/pal/src/file/file.cpp]: Delete comment?

**Comment by janvorli** [comment]
Fixed the nits from the PR and amended the commit.

**Pull request merged**